src/cli/search.rs

```diff
@@ -99,11 +100,155 @@ pub async fn execute(args: Args) -> miette::Result<()> {
     let repo_data = fetch_sparse_repodata(&channels, &platforms).await?;
 
     let p = PackageName::try_from(package_name.clone()).into_diagnostic()?;
+
+    if contains_wildcard(&package_name) {
+        let result = search_package_by_wildcard(p, &package_name, repo_data, limit).await;
+        return match result {
+            Ok(_) => Ok(()),
+            Err(report) => Err(report),
+        };
+    }
+
+    let result = search_exact_package(p, &package_name, repo_data).await;
+    return match result {
+        Ok(_) => Ok(()),
+        Err(report) => Err(report),
+    };
+}
+
+fn contains_wildcard(string: &str) -> bool {
```

Comment:
Nitpick: I would only put this in a function when it is used multiple times. In this case its just another click to see what the wildcard is.